test/ui/widgets/language_selector_dropdown_test.dart

```diff
@@ -0,0 +1,261 @@
+/// ------------------------------------------------------------------
+class FakeLocalizationNotifier extends LocalizationNotifier {
+  FakeLocalizationNotifier({
+    required String initialSelectedLanguage,
+    Map<String, String>? supported,
+  }) : _supportedLocales = supported ?? LocalizationService.supportedLocales {
+    final deviceLocaleCode = LocalizationService.getDeviceLocale();
+
+    state = LocalizationState(
+      currentLocale: Locale(
+        initialSelectedLanguage == 'system' ? deviceLocaleCode : initialSelectedLanguage,
+      ),
+      selectedLanguage: initialSelectedLanguage,
+    );
+  }
+
+  final Map<String, String> _supportedLocales;
+
+  String? lastChangedLocale;
+  int changeLocaleCallCount = 0;
+
+  @override
+  Map<String, String> get supportedLocales => _supportedLocales;
+
+  @override
+  Future<bool> changeLocale(String localeCode) async {
+    changeLocaleCallCount++;
+    lastChangedLocale = localeCode;
+
+    final deviceLocaleCode = LocalizationService.getDeviceLocale();
+
+    state = state.copyWith(
+      selectedLanguage: localeCode,
+      currentLocale: Locale(
+        localeCode == 'system' ? deviceLocaleCode : localeCode,
+      ),
+      isLoading: false,
+      error: null,
+    );
+
+    // Always report success in the fake
+    return true;
+  }
+}
+
+/// ------------------------------------------------------------------
+/// Test wrapper that sets up ProviderScope + ScreenUtil + MaterialApp
+/// and overrides `localizationProvider` with our FakeLocalizationNotifier.
+/// ------------------------------------------------------------------
+class WidgetTestHelper extends StatelessWidget {
+  final Widget child;
+  final FakeLocalizationNotifier fakeNotifier;
+
+  const WidgetTestHelper({
+    super.key,
+    required this.child,
+    required this.fakeNotifier,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ProviderScope(
+      overrides: [
+        localizationProvider.overrideWith((ref) => fakeNotifier),
+      ],
+      child: ScreenUtilInit(
+        designSize: const Size(390, 844),
+        builder: (_, _) {
+          return MaterialApp(
+            home: Scaffold(
+              body: Padding(
+                padding: const EdgeInsets.all(16),
+                child: child,
+              ),
+            ),
+          );
+        },
+      ),
+    );
+  }
+}
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  const fakeTranslationsJson = '''
+{
+  "shared": {
+    "system": "System"
+  },
+  "settings": {
+    "title": "Settings",
+    "subtitle": "Configure your app"
+  },
+  "greeting": "Hello {name}"
+}
+''';
+
+  setUpAll(() {
+    final jsonEncoded = utf8.encode(fakeTranslationsJson);
+    final jsonByteData = ByteData.view(Uint8List.fromList(jsonEncoded).buffer);
+
+    const tinySvg = '<svg xmlns="http://www.w3.org/2000/svg" width="1" height="1"></svg>';
```

Comment:
Again, I had commented it before, but this tiny svg logic seems to be unnecessary. I commented all this setUpApp block and the tests passed without it (115-145)